Simplify `$anonymousPattern` further

lucaswerkmeister opened the pull request:
No need to match the rest of the string, or for the regex to be case-insensitive now that it doesn’t contain letters anymore.

Once we’re on PHP 8.0+, we can even use `str_starts_with()` (but currently we still seem to be running on PHP 7.3 AFAICT).

lucaswerkmeister committed:
Simplify $anonymousPattern further

No need to match the rest of the string, or for the regex to be
case-insensitive now that it doesn’t contain letters anymore. In fact,
now that we’re on PHP 8, we can even use str_starts_with() and avoid the
regex altogether.

Ladsgroup commented:
> but currently we still seem to be running on PHP 7.3 AFAICT.

I think I fixed that while deploying the previous PR. Wanna use str_starts_with?

lucaswerkmeister commented:
Ah, then I was misled by `webservice status` ([T312278](https://phabricator.wikimedia.org/T312278)). I’ll update the PR.

lucaswerkmeister committed:
Update service.template

Bump PHP to 8.4; drop `canonical` (so outdated that `webservice` warns
about it now); and drop `backend` (kubernetes is the only option now).

lucaswerkmeister commented:
Updated and included an update to `service.template` too.
